src/edgepi/calibration/edgepi_eeprom.py

```diff
@@ -195,8 +195,17 @@ def __parameter_sanity_check(self, mem_addr: int = None,
             length (int): length of data to read
             user_space (bool): True if mem_addr is in user space
         """
-        mem_total_size = (EdgePiMemoryInfo.USER_SPACE_END_BYTE.value+1) if user_space \
-                      else (EdgePiMemoryInfo.USER_SPACE_START_BYTE.value)
+        # Max allowable memory size = End address - Start address + 1
+        user_max_size = EdgePiMemoryInfo.USER_SPACE_END_BYTE.value -\
+                        EdgePiMemoryInfo.USER_SPACE_START_BYTE.value +1
+        private_max_size = EdgePiMemoryInfo.PRIVATE_SPACE_END_BYTE.value -\
+                            EdgePiMemoryInfo.PRIVATE_SPACE_START_BYTE.value +1
+
+        # subtract the user space offset if user_space is set
+        mem_addr = mem_addr-EdgePiMemoryInfo.USER_SPACE_START_BYTE.value if user_space else mem_addr
+        # select mem max size
+        mem_total_size = user_max_size if user_space else private_max_size
+
```

Comment:
looking at this function again, I am not sure why do we even need to calculate `mem_total_size`. imagine you remove all this:
```
        # Max allowable memory size = End address - Start address + 1
        user_max_size = EdgePiMemoryInfo.USER_SPACE_END_BYTE.value -\
                        EdgePiMemoryInfo.USER_SPACE_START_BYTE.value +1
        private_max_size = EdgePiMemoryInfo.PRIVATE_SPACE_END_BYTE.value -\
                            EdgePiMemoryInfo.PRIVATE_SPACE_START_BYTE.value +1

        # subtract the user space offset if user_space is set
        mem_addr = mem_addr-EdgePiMemoryInfo.USER_SPACE_START_BYTE.value if user_space else mem_addr
        # select mem max size
        mem_total_size = user_max_size if user_space else private_max_size
```
then simply isn't it enough to have the following:
```
        last_addressable_byte = EdgePiMemoryInfo.USER_SPACE_START_BYTE.value if user_space else EdgePiMemoryInfo.PRIVATE_SPACE_END_BYTE.value

        # Checks whether proper values are passed
        if mem_addr is None or length is None or mem_addr < 0 or length <= 0:
            raise ValueError(f'Invalid Value passed: {mem_addr}, {length}')
        # Checks whether starting address and length of data to read/write are within memory bound
        if mem_addr + length -1 > last_addressable_byte:
            raise MemoryOutOfBound(f"Operation range is over the size of the memory by "
                                   f"{mem_addr + length -1 - last_addressable_byte}")
```

doing so you can avoid the assumption you are making in the following that user space is after private space:
```
        # subtract the user space offset if user_space is set
        mem_addr = mem_addr-EdgePiMemoryInfo.USER_SPACE_START_BYTE.value if user_space else mem_addr
```

I am just trying to push the code such that we do not make any hard coded assumption about how user and private spaces are configured so that in future if we need to change something, we wouldn't need to change anything, we just modify the start and end addresses for each space and the code itself does not make any other assumption about the location of each space.

Reply:
6f3fdc82f6d385e6aff750d06081ee72429cb679

I see, yes. If we check using the last address it handles the assumption I have about the memory location. Change's been tested and pushed